compiler/rustc_middle/src/lint.rs

```diff
@@ -483,7 +483,10 @@ pub fn in_external_macro(sess: &Session, span: Span) -> bool {
             DesugaringKind::ForLoop | DesugaringKind::WhileLoop | DesugaringKind::OpaqueTy,
         ) => false,
         ExpnKind::AstPass(_) | ExpnKind::Desugaring(_) => true, // well, it's "external"
-        ExpnKind::Macro(MacroKind::Bang, _) => {
+        ExpnKind::Macro(MacroKind::Bang, id) => {
+            if id.as_str() == "core::arch::global_asm" {
+                return false;
+            }
```

Comment:
Please add a comment explaining that this is needed such that the `unsafe_code` lint works correctly.

Also, what if an external macro generates `global_asm`? Would that be denied by the compiler?